### DISCUSSION TITLE
Remove item from cache

### DISCUSSION THREAD

**Discussion by apo1967** [Akryum/vue-apollo#1203] [Q&A]
Applies to vue-apollo 3.0.0-beta.11 , Vue 2

I am currently trying to figure out how to remove an item from cache. I have a mutation which deletes an entity and along with that some referenced entities. But I noticed these entities are not removed from the cache. Therefore when rendering another component that uses the "cache-and-network" fetch policy it fetches stale entities from cache first and does some requests to fetch details for these entities, which result in errors.

So what I tried is to read the fragments from the cache (which works) and then set it to `null` or empty. But that does not remove the items from the cache. So further research on the issue led me to the conclusion, that the only way currently to manually remove these items from cache would be to use `cache.evict(...)` as suggested in this [question on stackoverflow](https://stackoverflow.com/questions/63192774/apollo-client-delete-item-from-cache) or to remove the item by iterating over all items in the cache as suggested [in this article](https://medium.com/rbi-tech/tips-and-tricks-for-working-with-apollo-cache-3b5a757f10a0) but that does not seem to be at hand with vue apollo.

So my question is if this problem can be solved with vue apollo in order to not having to use the fetch policy "network-only" which would take away the benefit of caching?

**Comment by Akryum** [comment]
> but that does not seem to be at hand with vue apollo

Why?

**Reply by apo1967** [reply]
Ok, so first of all sorry for misusing the issue board. Must have missed the link to discussions.

Could you please give me a hint on how to access the cache evict method or provide me with a link to the appropriate documentation? So from my understanding when I have an update handler it will inject the store (aka cache?!) like 
`update(store, …)`
Navigating to the source I can see it provides methods for read/write queries and fragments, but other than that I can not see (possibly due to my lack of javascript skills) how get further access to the cache.

Thanks in advance,
apo

**Reply by Akryum** [reply]
```js
await this.$apollo.mutate({
  mutation: gql`...`,
  update: (store) => {
    store.evict({ id: store.identify({ __typename: 'Message', id: 'abc' }) })
  },
})
```

**Reply by Akryum** [reply]
See [mutate](https://www.apollographql.com/docs/react/api/core/ApolloClient/#ApolloClient.mutate)

**Reply by apo1967** [reply]
Thank you very much!
